stl/inc/expected

```diff
@@ -722,6 +1117,23 @@ public:
     }
 
 private:
+    // These overloads force copy elision from the invoke call into _Value
+    template <class _Fn, class _Ux>
+    constexpr expected(_Construct_expected_from_invoke_result_tag, _Fn&& _Func, _Ux&& _Arg) noexcept(
+        noexcept(_Ty(_STD invoke(_STD forward<_Fn>(_Func), _STD forward<_Ux>(_Arg)))))

```

Comment:
I do not love that this is a C cast, but also it's not in code that gets run, and I believe the noexcept-ness of a C cast is always equivalent to a `static_cast` that also compiles...

Reply:
+1 for using `static_cast` here, and in the overloads below, also for `_Err`.